ecosystem/gstreamer_plugin/gstmtltxsink.c

```diff
@@ -0,0 +1,653 @@
+static gboolean gst_mtltxsink_parse_fps(GstVideoInfo* info, enum st_fps* fps);
+static struct st_frame* gst_mtltxsink_get_frame(GstMtlTxSink* sink);
+
+static gboolean gst_mtltxsink_parse_input_fmt(GstVideoInfo* info,
+                                              enum st_frame_fmt* fmt) {
+  GstVideoFormatInfo* finfo = info->finfo;
+
+  if (finfo->format == GST_VIDEO_FORMAT_v210) {
+    *fmt = ST_FRAME_FMT_V210;
+  } else if (finfo->format == GST_VIDEO_FORMAT_I420_10LE) {
+    *fmt = ST_FRAME_FMT_YUV422PLANAR10LE;
+  } else {
+    return FALSE;
+  }
+
+  return TRUE;
+}
+
+static gboolean gst_mtltxsink_parse_fps(GstVideoInfo* info, enum st_fps* fps) {
+  gint fps_div;
+  if (info->fps_n == 0 || info->fps_d == 0) {
+    return FALSE;
+  }
+
+  fps_div = info->fps_n / info->fps_d;
+
+  switch (fps_div) {
+    case 24:
+      *fps = ST_FPS_P24;
+      break;
+    case 25:
+      *fps = ST_FPS_P25;
+      break;
+    case 30:
+      *fps = ST_FPS_P30;
+      break;
+    case 50:
+      *fps = ST_FPS_P50;
+      break;
+    case 60:
+      *fps = ST_FPS_P60;
+      break;
+    case 120:
+      *fps = ST_FPS_P120;
+      break;
+    default:
+      return FALSE;
+  }
+
+  return TRUE;
+}
+
+static void gst_mtltxsink_class_init(GstMtlTxSinkClass* klass) {
+  GObjectClass* gobject_class;
+  GstElementClass* gstelement_class;
+  GstVideoSinkClass* gstvideosinkelement_class;
+
+  gobject_class = G_OBJECT_CLASS(klass);
+  gstelement_class = GST_ELEMENT_CLASS(klass);
+  gstvideosinkelement_class = GST_VIDEO_SINK_CLASS(klass);
+
+  gst_element_class_set_metadata(
+      gstelement_class, "MtlTxSt20Sink", "Sink/Video",
+      "Mtl transmition plugin for st2110 standard rawvideo st20",
+      "Dawid Wesierski <dawid.wesierski@intel.com>");
+
+  gst_element_class_add_static_pad_template(gstelement_class,
+                                            &gst_mtltxsink_sink_pad_template);
+
+  gobject_class->set_property = GST_DEBUG_FUNCPTR(gst_mtltxsink_set_property);
+  gobject_class->get_property = GST_DEBUG_FUNCPTR(gst_mtltxsink_get_property);
+  gstvideosinkelement_class->parent_class.start = GST_DEBUG_FUNCPTR(gst_mtltxsink_start);
+  gstvideosinkelement_class->parent_class.stop = GST_DEBUG_FUNCPTR(gst_mtltxsink_stop);
+
+  g_object_class_install_property(
+      gobject_class, PROP_SILENT,
+      g_param_spec_boolean("silent", "Silent", "Produce verbose output ?", FALSE,
+                           G_PARAM_READWRITE));
+
+  g_object_class_install_property(
+      gobject_class, PROP_TX_DEV_ARGS_PORT,
+      g_param_spec_string("dev-port", "Dpdk device port",
+                          "DPDK port for synchronous ST2110 ST20 raw video "
+                          "transmission, bound to the VFIO DPDK driver. ",
+                          NULL, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property(
+      gobject_class, PROP_TX_DEV_ARGS_SIP,
+      g_param_spec_string("dev-ip", "Local device ip",
```

Comment:
```suggestion
      g_param_spec_string("dev-ip", "Local device IP",
```

Reply:
IP address

Reply:
Hmmm I dont think the Local device IP address would be better for the short description 
The long description already has the full phrase